ext/curl/url.c

```diff
@@ -0,0 +1,196 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Author: Pierrick Charron <pierrick@php.net>                          |
+   +----------------------------------------------------------------------+
+*/
+
+#define ZEND_INCLUDE_FULL_WINDOWS_HEADERS
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "php.h"
+#include "Zend/zend_exceptions.h"
+
+#include "curl_private.h"
+
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+
+/* CurlUrl class */
+zend_class_entry *curl_CURLUrl_ce;
+zend_class_entry *curl_CURLUrlException_ce;
+
+static zend_result php_curl_url_set(php_curlurl *uh, zend_long option, const char *str, const size_t len, unsigned int flags)
+{
+	CURLUcode error = curl_url_set(uh->url, option, str, flags);
+	SAVE_CURL_ERROR(uh, error);
+
+	return error == CURLUE_OK ? SUCCESS : FAILURE;
+}
+
+static void php_curl_url_ctor(INTERNAL_FUNCTION_PARAMETERS)
+{
+	php_curlurl *uh;
+	zend_string *url = NULL;
+
+	ZEND_PARSE_PARAMETERS_START(0,1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(url)
```

Comment:
```suggestion
		Z_PARAM_PATH_OR_NULL(url)
```

This automatically checks for null bytes

Reply:
I was thinking about using it, just the name feels wrong ! Anyway I did the change and it works fine ! Thanks